### ISSUE TITLE
Update ``exec-used`` documentation

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#7039]
### Bug description

In the following code, what can be use in ``exec`` is properly restricted. 

```python
def get_user_code(name):
    return input(f'Enter code to be executed please, {name}: ')


username = "Ada"
allowed_globals = {'__builtins__' : None}
allowed_locals = {'print': print}
exec(get_user_code(username), allowed_globals, allowed_locals)
```

For example, if Ada is reasonable:
```
Enter code to be executed please, Ada: print("The more I study, the more insatiable do I feel my genius for it to be")
The more I study, the more insatiable do I feel my genius for it to be
```

But if she's trying to read all our env variable:
```
Enter code to be executed please, Ada: import os;print(os.environ)
Traceback (most recent call last):
  File "c.py", line 8, in <module>
    exec(get_user_code(username), globals, locals)
  File "<string>", line 1, in <module>
SystemError: ../Objects/dictobject.c:1438: bad argument to internal function
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:2:11: W0141: Used builtin function 'input' (bad-builtin)
a.py:8:0: W0122: Use of exec (exec-used)
```

### Expected behavior

No ``exec-used``, proper care were taken.

### Pylint version

```shell
2.15.0-dev0
```

**Comment by DanielNoord** [comment]
What would be the required change: no longer emit `exec` if `globals` and/or `locals` is provided?

**Comment by Pierre-Sassoulas** [comment]
> What would be the required change: no longer emit exec if globals and/or locals is provided?

Yes

**Comment by DanielNoord** [comment]
Kind of depends on how "safe" we want to keep these.
If you're using Python for a server or API backend I think it would be good to have this warning irrespective of `globals` and `locals`. 
`exec("raise SystemExit", {}, {})` could still cause many troubles.

**Comment by Pierre-Sassoulas** [comment]
Maybe we could burst the message into "unsafe-exec-used" and "exec-used" ?

**Comment by DanielNoord** [comment]
I'm not sure. In the example you gave above I can still `raise SystemExit`. There are probably other attack vectors that we don't even know about.
If you really want to `exec`, just disable the message? It's a short message name anyway 😄

**Comment by Pierre-Sassoulas** [comment]
Being able to raise exception is not the same as being able to read all the env variable and transferring them to a server using ``requests`` though. Imo if ``globals`` / ``locals`` are defined, it probably means everything we'll raise about this call will be a false positive because the user thought about it.

**Comment by DanielNoord** [comment]
Try ``exec(input(), {}, {})`` and type in:
`exec("""\nwith open("file.txt", "w", encoding="utf-8") as file:\n    file.write("test")\n""")`

I can write to your system, I can read from your system and then print the contents, I can probably even import `requests` and do it that way. `exec` should always be warned against imo. Just to make sure users are very sure they want to use it and make sure it can't be controlled by anybody else.

**Comment by Pierre-Sassoulas** [comment]
Well that is convincing. Let's update the doc about ``exec-used`` with details from this conversation and close this.
